### PR TITLE
Chores/paged segments

### LINKED ISSUE #1575: Listing segments for a project can be slow for large amounts of segments

### PR DESCRIPTION
**Pull request by kyle-ssg**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/deployment/locally-api#pre-commit) to check linting
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?

## Changes

In order to support segment paging, this PR removes the reliance of the segment list for areas of the application and separates the segment override select from the segment list store.

### CODE REVIEW

File: frontend/web/components/SegmentOverrides.js [hunk 1]
```diff
@@ -64,12 +64,12 @@ const SegmentOverrideInner = class Override extends React.Component {
...
                     >
                         <strong>
-                            {name || v.segment.name}
-                            {segment && segment.feature && (
-                            <div className="unread ml-2 px-2">
-                                        Feature-Specific
-                            </div>
-                            )}
+                            {name || v.segment_name}
+                            {/*{segment && segment.feature && (*/}
```
Comment (kyle-ssg) [inline frontend/web/components/SegmentOverrides.js]: @matthewelwell If we can add the segment_feature we can re-add this

File: frontend/common/providers/withSegmentOverrides.js [hunk 2]
```diff
@@ -1,23 +1,18 @@
...
-
+import FeatureListStore from '../stores/feature-list-store'
 export default (WrappedComponent) => {
     class HOC extends React.Component {
         static displayName = 'withFoo';
 
         constructor(props) {
             super(props);
             ES6Component(this);
+            this.listenTo(FeatureListStore, 'saved', () => {
```
Comment (kyle-ssg) [inline frontend/common/providers/withSegmentOverrides.js:11]: Fixes a current live issue where saving after removing a segment override fails

File: frontend/web/components/SegmentFeatureOverrides.js [hunk 3]
```diff
@@ -344,7 +343,7 @@ class TheComponent extends Component {
                             />
                         </Flex>
                     )}
-                    {value && !!value.length && (
+                    {visibleValues && !!visibleValues.length && (
```
Comment (kyle-ssg) [inline frontend/web/components/SegmentFeatureOverrides.js:346]: Fixes a small issue where the UI showed segment priority dragging instructions where there were no overrides visible

File: frontend/web/components/SegmentOverrides.js [hunk 4]
```diff
@@ -355,54 +357,35 @@ class TheComponent extends Component {
...
+                                data-test="select-segment"
+                                placeholder="Create a Segment Override..."
+                                filter={filter}
+                                value={this.state.selectedSegment}
+                                onChange={selectedSegment => this.setState({ selectedSegment }, this.addItem)}
                             />
                         </Flex>
                     )}
-                    {Utils.getFlagsmithHasFeature('flag_based_segments') && !this.props.showCreateSegment && !this.props.readOnly && (
+                    {Utils.getFlagsmithHasFeature('flag_based_segments') && !this.props.showCreateSegment && !this.props.readOnly && !this.props.disableCreate &&  (
```
Comment (kyle-ssg) [inline frontend/web/components/SegmentOverrides.js:388]: Fixes a current issue where "Create Feature-Specific Segment" shows in the segment override list for a segment's associated features.

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**